d3a_interface/sim_results/bills.py

```diff
@@ -250,13 +254,38 @@ def _accumulate_grid_fee_charged(self, area_dict, area_core_stats):
             self._accumulate_grid_fee_charged(child, area_core_stats)
 
     def update(self, area_dict, area_core_stats):
+        # Updates the self.market_fees dict which keeps track of the accumulated market fees for
+        # each area.
         self._update_market_fees(area_dict, area_core_stats)
+        # Calculates the cumulative_fee_charged_per_market , which in reality is the sum of the
+        # grid fees for all markets for the whole simulation duration.
         self._accumulate_grid_fee_charged(area_dict, area_core_stats)
```

Comment:
I just had a look into the code of both functions an AFAIU, they are doing the same. 
If you agree, i suggest to get rid of one of them.

Reply:
I assume you are talking about the methods `_update_market_fees` and `_accumulate_grid_fee_charged`. Unfortunately they do not do the same thing, _accumulate_grid_fee_charged is actually calculating a sum of all market fees of all areas as a single number (therefore it only changes a float value), and the _update_market_fees maintains a dict with a mapping of area_uuid -> accumulated grid fees for that area for the whole simulation.
What could be improved here is that we could avoid the second recursion over the area_dict and calculate in one recursion both stats. If you agree I'll do this change.

Reply:
Ah, thanks a lot for the explanation, i looked again and was probably confused by the naming of `cumulative_fee_charged_per_market` in the `_accumulate_grid_fee_charged` which should be named rather `cumulative_fee_charged_whole_sim` or similar (but please skip this if you can not bother to open a d3a-web PR for that)
I agree with you proposal to only loop once and combine the functions. Thanks a lot!

Reply:
I think you are right, it is better to fix this now and name it properly, otherwise more people will be confused here. Will create a d3a-web PR to change the usage of this variable